public/react-services/saved-objects.js

```diff
@@ -126,6 +145,23 @@ export class SavedObject {
     }
   }
 
+  /**
+   *
+   * Given an index pattern ID, checks if it exists
+   */
+  static async getExistingIndexPattern(patternID) {
+    try {
+      const result = await GenericRequest.request(
+        'GET',
+        `/api/saved_objects/index-pattern/${patternID}?fields=title&fields=fields`
+      );
+
+      return result.data;
+    } catch (error) {
+      return ((error || {}).data || {}).message || false ? error.data.message : error.message || false;
+    }
+  }
```

Comment:
Calling this method always returns a resolved promise, I am not sure if this could affect to another logic